pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -33,8 +33,8 @@ import (
 	"k8s.io/klog/v2"
 
 	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	cdiv1cloneauth "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/clone"
```

Comment:
I think `kubevirt.io/containerized-data-importer-api/pkg/clone` would be better.  Let's keep `pkg/apis` for kubernetes apis

Reply:
So we already have https://github.com/kubevirt/containerized-data-importer/blob/be1add08d51ebba2f62538afe6d8c13db952d712/pkg/controller/common/util.go#L45

Reply:
makes sense, was about to do this, but then realized we have https://github.com/kubevirt/containerized-data-importer/issues/2552 which will accept a DV

Reply:
okay then maybe it makes to refactor to do something like:

```golang
ok, err := dv.Authorize()
```

Reply:
so `Authorize()` will go in ${version}/utils, and we will be able to have different authorization logic for different version?
I am just making sure, I have no issue with moving to `kubevirt.io/containerized-data-importer-api/pkg/clone`

Reply:
I guess what I'm saying is do #2552 in this pr and have `Authorize` be a DataVolume member function.

Reply:
Posted a first draft just to make sure we are fine with the direction
- Still missing authorization via SA
- Feels like the logic is clunky, any pointers are greatly appreciated
- Would aggregating the proxies into one interface make sense?